### PR TITLE
UUI integration

### PR DESCRIPTION
**Pull request by kianzarrin**
user can optionally turn on:
- UUI's cross-mod hotkey conflict resolution for TMPE.
- Add TMPE button to UUI.

How it works:
- I do not check if user is subscribed to UUI. if user is not subscribed I just use the stand alone library.
- TMPE always uses UUI's cross-mod hot-key resolution. for this reason TMPE's activation key is handled by UUI (so that it can be suppressed).
- if the "Use UnifiedUI" option is enabled, TMPE icon will appear inside UUI. otherwise it will appear outside UUI. (in both cases UUI's cross-mod hotkey resolution is used.

how to test:
~~don't use UnifiedUI mod at all 
or download latest version from here: https://github.com/kianzarrin/UnifiedUI/releases
the WS version is too old (i will update).~~
I updated WS

[TMPE.zip](https://ci.appveyor.com/api/projects/krzychu124/tmpe/artifacts/TMPE.zip?branch=UUI)

### CODE REVIEW

**Comment by kianzarrin** [comment]
this is draft because I still need to add option to option menu.
and update TMPE texture

**Comment by krzychu124** [comment]
why `log-dir` is the target branch?

File: TLM/TLM/State/OptionsTabs/OptionsGeneralTab.cs [hunk 1]
```diff
@@ -82,6 +86,18 @@ public static class OptionsGeneralTab {
                                   defaultValue: GlobalConfig.Instance.Main.MainMenuPosLocked,
                                   eventCallback: OnLockMenuChanged) as UICheckBox;
 
+            _useUUI = generalGroup.AddCheckbox(
+                text: T("General.Checkbox:Use UnifedUI"),
```
Comment (kvakvs) [inline TLM/TLM/State/OptionsTabs/OptionsGeneralTab.cs:90]: Typo in `UnifiedUI` translation key, translation fix will probably be enough

File: TLM/TLM/UI/TrafficManagerTool.cs [hunk 2]
```diff
@@ -1962,5 +1968,34 @@ void DefaultRenderOverlay(RenderManager.CameraInfo cameraInfo)
                 OnscreenDisplay.DisplayIdle();
             }
         }
+        public void AddUUIButton() {
+            try {
+                var hotkeys = new UUIHotKeys { ActivationKey = KeybindSettingsBase.ToggleMainMenu.Key };
+                hotkeys.AddInToolKey(KeybindSettingsBase.ToggleTrafficLightTool.Key);
+                hotkeys.AddInToolKey(KeybindSettingsBase.LaneArrowTool.Key);
+                hotkeys.AddInToolKey(KeybindSettingsBase.LaneConnectionsTool.Key, () => Options.laneConnectorEnabled);
```
Comment (kvakvs) [inline TLM/TLM/UI/TrafficManagerTool.cs:1976]: Names on the options are ambiguous
We have switches in TM:PE which enable entire feature
And also we have hotkeys which don't toggle the feature, but they activate tool button
Change names for hotkeys which activate tools from `***Enabled` to for example `Use***Tool`

Comment (kianzarrin) [reply TLM/TLM/UI/TrafficManagerTool.cs:1976]: it is out of scope of this PR. you should create an issue.

**Review by DaEgi01 — APPROVED** [review]

**Review by krzychu124 — CHANGES_REQUESTED** [review]
Everything seem to be working in-game, I left one question in the code.

File: TLM/TLM/State/ConfigData/Main.cs [hunk 3]
```diff
@@ -21,6 +21,11 @@ public class Main {
         public int MainMenuY = MainMenuWindow.DEFAULT_MENU_Y;
         public bool MainMenuPosLocked = false;
 
+        /// <summary>
+        /// Put button inisde UUI and use the UUI Hotkey conflict resolution.
+        /// </summary>
+        public bool UseUUI = true;
```
[the remark this answers is not in the document]
Comment (kianzarrin) [reply TLM/TLM/State/ConfigData/Main.cs:27]: good point

**Comment by krzychu124** [comment]
@kianzarrin hehe, you requested review but didn't change anything 😄 
Anyways, I fixed Appveyor build pipeline to include `UnifiedUILib.dll` in the zip file

**Review by kvakvs — APPROVED** [review]

File: TLM/TLM/State/ConfigData/Main.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (krzychu124) [reply TLM/TLM/State/ConfigData/Main.cs:27]: @kianzarrin still waiting for change 😓 That's the only thing holding me from approving the PR 😉

[the remark this answers is not in the document]
Comment (kianzarrin) [reply TLM/TLM/State/ConfigData/Main.cs]: done :)

**Review by krzychu124 — APPROVED** [review]
Cool 👍

**Pull request merged**